### DISCUSSION TITLE
Questions about integrating cloud messaging.

### DISCUSSION THREAD

**Discussion by kimmy-wang** [firebase/flutterfire#5819] [Q&A]
1. Cloud messages cannot be received on the Android emulator and real device, but cloud messages can be received on the real iOS device. What is causing this?
2. After adding the method `requestPermission` to the code, a popup window for requesting permission appears on the iOS simulator, but it does not appear on the Android simulator.

Regarding the second question, after reading the source code, I found that `On Android, is it not required to call this method`.

**Comment by Anoirwork** [comment]
Have you verified your SHA1 and SHA256 ? Android with fcm is bit pluzzing at this point.
This was my tought hope this helps.
Have a great day.
